### PR TITLE
docs: update oep-49 for clarity around new apps

### PR DESCRIPTION
**Pull request by rgraber**
Issue: https://github.com/openedx/edx-platform/issues/31659

Update OEP-49 by 

- moving it to Best Practices
- adding a section for the ADR that should be included
- adding a change history

Note this will be kept as 2 separate commits for merge for cleaner history.

### CODE REVIEW

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (robrap) [inline oeps/best-practices/oep-0049-django-app-patterns.rst:273]: Can you add a link to https://open-edx-proposals.readthedocs.io/en/latest/best-practices/oep-0019-bp-developer-documentation.html#adrs?

**Review by robrap — COMMENTED** [review]
Also, I didn't see you post this review in the community channels. Were you planning to do that after you have internal approval?

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:270]: Actually that's an interesting point. If a new repo has only one new app, do we want both a purpose of this app and a purpose of this repo?

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst]: [fixed]

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (kdmccormick) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:270]: My gut reaction is that requring separate ADRs for the repo *and* the app sounds like too much paperwork. However, requiring ADRs for apps created in exisitings repo does sound fair to me.

As a conpromise, we could recommend that the initial app(s) have stub ADRs, each of which point to the repo-level ADR. The repo cookiecutter could be updated to provide that.

Comment (timmc-edx) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:270]: Yeah, I'd lost track of this being for apps in general, which might be in their own repos. If it's a single-app repo, I think there should just be one docs/ dir, either at the repo root or inside the app. Top-level sounds more convenient and easy to notice. The one reason I can think of to put it inside the app is in case the code moves into another repo (maybe absorbed into an IDA?), since having it at the repo root might risk it being lost. But I don't know how plausible that is.

Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:270]: I think @kdmccormick 's suggestion of the stub ADR pointing to the main ADR addresses most of your concerns @timmc-edx . The documentation is effectively centralized, but if someone moves they app folder they should notice that something is up when their link breaks.

Comment (robrap) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:270]: I like that proposal as well.

Comment (timmc-edx) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:270]: A stub makes to me as well, yeah.

Comment (kdmccormick) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:270]: For what it's worth, I wrote that proposal thinking about apps in an IDA, not single-app packages.

I would agree with @timmc-edx's earlier point that a single ADR in the root dir of the a single-app package is probably good enough. But if that's not that the consensus that's been reached here, that's fine with me too.

Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst]: Updated with the stub-ADR-in-the-app version. Even I find the phrasing I used confusing so if anyone has ideas for ways to express this more clearly I'm very much open to them.

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (robrap) [inline oeps/best-practices/oep-0049-django-app-patterns.rst:275]: 1. I stick with my original proposal to update the title of this section to `docs/decisions/0001-purpose-of-this-app.rst`. All of the other sections are relative to the app directory.
2. I'm wondering if this could be simpler, and not necessarily require (0001-purpose-of-this-repo.rst), since this OEP isn't about that. How do you feel about something like the following?
```
If this is the only app in the entire repo, this ADR could refer to a ``0001-purpose-of-this-repo.rst`` in the repository-level docs/decisions directory.
```

Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:275]: I might replace "could" with "should" but that's fine by me

Comment (robrap) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:275]: @rgraber: optional: You could reference https://github.com/openedx/edx-cookiecutters#3-todos-after-running-cookiecutter if you'd like.

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (robrap) [reply oeps/best-practices/oep-0049-django-app-patterns.rst]: [inform] The updated text is in https://github.com/openedx/open-edx-proposals/pull/455/files#r1122281928, but the PR hasn't been updated yet.

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (robrap) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:275]: I approve once this has been updated.

Comment (kdmccormick) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:275]: I recommend "[could/should] be a stub that links to" over "should refer to". I think the latter leaves room for misintepretation (ie "do I have to write a whole new ADR, that happens to contain a link to the repo ADR?")

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst]: PR has been updated

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (kdmccormick) [inline oeps/best-practices/oep-0049-django-app-patterns.rst:275]: Rather than adding this to the post-cookiecutter TODO list, could we add it to the cookiecutters as a part of this OEP change?

Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:275]: As in postpone this update until we have updated the cookiecutter?

Comment (kdmccormick) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:275]: Yeah, or at least until we have a draft PR ready to go for updating the cookiecutter. Since we are asking folks to make these additional ADRs starting now, my intuition is that it'd be best to provide the cookiecutter scaffolding now as well.

I don't know, maybe I'm underestimating the effort it'd take to update the cookiecutter. Don't take this as a blocking suggestion; what you have here is OK with me.

**Review by kdmccormick — APPROVED** [review]
I have a suggestion in [this thread](https://github.com/openedx/open-edx-proposals/pull/455/files#r1123568822), but it's not blocking. Thanks for the update!

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:275]: I'm nervous about blocking this, so I'll leave this as is and then make an issue to add this to the cookiecutter if there isn't one already. If there is one I'll add a task to update this line.

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (robrap) [inline oeps/best-practices/oep-0049-django-app-patterns.rst:273]: Nit: Missing period at the end.

File: oeps/best-practices/oep-0049-django-app-patterns.rst
Comment (rgraber) [reply oeps/best-practices/oep-0049-django-app-patterns.rst:275]: This is now https://github.com/openedx/edx-cookiecutters/issues/299. We can further discuss there.

**Pull request closed without merging**
